state-sync/inter-component/mempool-notifications/src/lib.rs

```diff
@@ -36,27 +44,31 @@ pub trait MempoolNotificationSender: Send {
     ) -> Result<(), Error>;
 }
 
-/// The state sync component responsible for notifying mempool.
+/// This method returns a (MempoolNotifier, MempoolNotificationListener) pair that can be used
+/// to allow state sync and mempool to communicate.
 ///
-/// Note: When a MempoolNotifier instance is created, mempool must take and
-/// listen to the receiver in the corresponding MempoolNotificationListener.
+/// Note: state sync should take the notifier and mempool should take the listener.
+pub fn new_mempool_notifier_listener_pair() -> (MempoolNotifier, MempoolNotificationListener) {
```

Comment:
Okay, so you're adopting the `diem_channel` approach

Reply:
Yeah, it seems a little nicer :)